payments_payu/provider.py

```diff
@@ -43,16 +43,20 @@
 
 
 def add_extra_data(payment, new_extra_data):
+    payment.refresh_from_db()
     if payment.extra_data:
         old_extra_data = json.loads(payment.extra_data)
     else:
         old_extra_data = {}
     extra_data = {**old_extra_data, **new_extra_data}
     payment.extra_data = json.dumps(extra_data, indent=2)
-    payment.save()
+    payment._change_reason = 'PayU Payments: add_extra_data'  # For django-simple-history
+    payment.save(update_fields=["extra_data"])
+    payment._change_reason = None
 
 
 def add_new_status(payment, new_status):
+    payment.refresh_from_db()
```

Comment:
Wouldn't this be better?

```suggestion
    payment.refresh_from_db(fields=["extra_data"])
```

Just to make it more future-proof...